### PR TITLE
MAINT: remove support for meta children and 2d pandas

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Addresses #789 and #913

Removes the support for storing pandas DataFrames within other DataFrames, including
- the `pysat_testing2d` instrument
- children in Meta objects

## Type of change

- Breaking change (fix or feature that would cause existing functionality
      to not work as expected)
- This change requires a documentation update

# How Has This Been Tested?

Tested against other pysat packages (ran vs this branch and pysat develop to see if new errors occur)
- [x] pysatCDAAC -- passes
- [x] pysatMadrigal -- no new errors vs develop
- [x] pysatMissions -- passes
- [x] pysatModels -- no new errors vs develop
- [x] pysatNASA -- no new errors vs develop
- [x] pysatSeasons -- passes with changes at https://github.com/pysat/pysatSeasons/pull/40
- [x] pysatSpaceWeather -- no new errors vs develop

**Test Configuration**:
* Operating system: Monterrey
* Version number: Python 3.10.9

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

If this is a release PR, replace the first item of the above checklist with the release
checklist on the wiki: https://github.com/pysat/pysat/wiki/Checklist-for-Release

### CODE REVIEW

File: pysat/_meta.py [hunk 1]
```diff
@@ -605,43 +506,10 @@ def match_name(func, var_name, index_or_column):
...
-                try:
-                    new_child_index = match_name(
-                        self.attr_case_name, key[1],
-                        self[new_index].children.data.index)
-                except AttributeError:
-                    raise NotImplementedError(
-                        'Cannot retrieve child meta data from multiple parents')
-
-                new_name = match_name(self.attr_case_name, key[2],
+                # Assume this is a label name
```
Comment (aburrell) [inline pysat/_meta.py:509]:
```suggestion

                # Assume this is a label name
```

**Review by aburrell — COMMENTED** [review]
Some feedback as an intermediate review.

File: pysat/tests/test_meta.py [hunk 2]
```diff
@@ -1999,33 +1363,22 @@ def teardown_method(self):
...
-            except AttributeError:
-                raise AssertionError(
-                    "Couldn't update mutable property {:}".format(
-                        prop.__repr__()))
+        try:
+            # Pandas does not support dataframe equality
+            setattr(self.meta, 'data', pds.DataFrame())
+
+            # Test that data is empty
+            assert self.meta.data.empty, "`meta.data` not updated correctly"
```
Comment (aburrell) [inline pysat/tests/test_meta.py:1378]:
```suggestion
            assert self.meta.empty, "`meta.data` not updated correctly"
```

Updated test to increase coverage and also ensure `empty` is being updated.

**Review by aburrell — COMMENTED** [review]
`_meta.py` lines 401 and 442 are newly uncovered, recommend adding an instance where meta is setting bytes instead of a string.

Also have a new uncovered line in `io.py` on line 1615 that could be addressed with a test for `inst_to_netcdf` that has a DataFrame of DataFrames to trip the ValueError.

**Comment by jklenzing** [comment]
> `_meta.py` lines 401 and 442 are newly uncovered, recommend adding an instance where meta is setting bytes instead of a string.
> 

Not sure why these show up as newly uncovered. Neither are covered in `develop` according to coveralls.

**Comment by jklenzing** [comment]
> Also have a new uncovered line in `io.py` on line 1615 that could be addressed with a test for `inst_to_netcdf` that has a DataFrame of DataFrames to trip the ValueError.

Looking at this one, other errors get tripped before we get here since this format is not allowed. I think we can delete this line.

**Review by JonathonMSmith — COMMENTED** [review]
There are more lines uncovered that haven't been discussed, but I can't think of straightforward ways to test them

**Comment by aburrell** [comment]
@JonathonMSmith yeah, there are a few things that could be done, but since there aren't *new* uncovered lines, I think that's best addressed in another branch.

**Review by aburrell — APPROVED** [review]

**Pull request merged**